cmd/amp/storage.go

```diff
@@ -0,0 +1,176 @@
+			return storageList(AMP, cmd, args)
+		},
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(StorageCmd)
+	StorageCmd.AddCommand(storagePutCmd)
+	StorageCmd.AddCommand(storageGetCmd)
+	StorageCmd.AddCommand(storageDeleteCmd)
+	StorageCmd.AddCommand(storageListCmd)
+}
+
+// storagePut validates the input command line arguments and creates or updates storage key-value pair
+// by invoking the corresponding rpc/storage method
+func storagePut(amp *client.AMP, cmd *cobra.Command, args []string) (err error) {
+	switch len(args) {
+	case 0:
+		return errors.New("Must specify storage key and storage value")
+	case 1:
+		return errors.New("Must specify storage value")
+	case 2:
+		// OK
+	default:
+		return errors.New("Too many arguments")
+	}
+
+	k := args[0]
+	v := args[1]
+	request := &storage.PutStorage{Key: k, Val: v}
+	client := storage.NewStorageClient(amp.Conn)
+	reply, err := client.Put(context.Background(), request)
+	if err != nil {
+		fmt.Println("Key not found: " + k)
+		return nil
+	}
+	fmt.Println(reply.Val)
+	return nil
+}
+
+// storageGet validates the input command line arguments and retrieves storage key-value pair
+//by invoking the corresponding rpc/storage method
+func storageGet(amp *client.AMP, cmd *cobra.Command, args []string) (err error) {
+	if len(args) > 1 {
+		log.Fatal("Too many arguments - check again!")
+	} else if len(args) == 0 {
+		log.Fatal("Must specify storage key")
+	}
+	k := args[0]
+	if k == "" {
+		log.Fatal("Must specify storage key")
+	}
+
+	request := &storage.GetStorage{Key: k}
+
+	client := storage.NewStorageClient(amp.Conn)
+	reply, err := client.Get(context.Background(), request)
+	if err != nil {
+		fmt.Println("Key not found: " + k)
+		return nil
+	}
+	fmt.Println(reply.Val)
+	return nil
+}
+
+// storageDelete validates the input command line arguments and deletes storage key-value pair
+// by invoking the corresponding rpc/storage method
+func storageDelete(amp *client.AMP, cmd *cobra.Command, args []string) (err error) {
+	if len(args) > 1 {
+		log.Fatal("Too many arguments - check again!")
+	} else if len(args) == 0 {
+		log.Fatal("Must specify storage key")
+	}
+	k := args[0]
+	if k == "" {
+		log.Fatal("Must specify storage key")
+	}
+
+	request := &storage.DeleteStorage{Key: k}
+
+	client := storage.NewStorageClient(amp.Conn)
+	reply, err := client.Delete(context.Background(), request)
+	if err != nil {
+		fmt.Println("Key not found: " + k)
+		return nil
+	}
+	fmt.Println(reply.Val)
+	return nil
+}
+
+// storageList validates the input command line arguments and lists all the storage
+// key-value pairs by invoking the corresponding rpc/storage method
+func storageList(amp *client.AMP, cmd *cobra.Command, args []string) (err error) {
+	if len(args) > 0 {
+		log.Fatal("Too many arguments - check again!")
+	}
+	request := &storage.ListStorage{}
+	client := storage.NewStorageClient(amp.Conn)
+	reply, err := client.List(context.Background(), request)
+	if err != nil {
```

Comment:
The way you are using `err` in the function, you can just return (because `err` has already been assigned a value. Otherwise, don't use a named return value.